renderdoc/driver/d3d11/d3d11_shaderdebug.cpp

```diff
@@ -239,6 +240,1368 @@ static void FlattenVariables(const rdcarray<ShaderConstant> &constants,
+  if(resourceData.dim == RESOURCE_DIMENSION_TEXTURE1D)
+  {
+    textureDecl = "Texture1D";
+    texdim = 1;
+    offsetDim = 1;
+  }
+  else if(resourceData.dim == RESOURCE_DIMENSION_TEXTURE2D)
+  {
+    textureDecl = "Texture2D";
+    texdim = 2;
+    offsetDim = 2;
+  }
+  else if(resourceData.dim == RESOURCE_DIMENSION_TEXTURE2DMS)
+  {
+    textureDecl = "Texture2DMS";
+    texdim = 2;
+    offsetDim = 2;
+  }
+  else if(resourceData.dim == RESOURCE_DIMENSION_TEXTURE3D)
+  {
+    textureDecl = "Texture3D";
+    texdim = 3;
+    offsetDim = 3;
+  }
+  else if(resourceData.dim == RESOURCE_DIMENSION_TEXTURECUBE)
+  {
+    textureDecl = "TextureCube";
+    texdim = 3;
+    offsetDim = 3;
+    useOffsets = false;
+  }
+  else if(resourceData.dim == RESOURCE_DIMENSION_TEXTURE1DARRAY)
+  {
+    textureDecl = "Texture1DArray";
+    texdim = 2;
+    offsetDim = 1;
+  }
+  else if(resourceData.dim == RESOURCE_DIMENSION_TEXTURE2DARRAY)
+  {
+    textureDecl = "Texture2DArray";
+    texdim = 3;
+    offsetDim = 2;
+  }
+  else if(resourceData.dim == RESOURCE_DIMENSION_TEXTURE2DMSARRAY)
+  {
+    textureDecl = "Texture2DMSArray";
+    texdim = 3;
+    offsetDim = 2;
+  }
+  else if(resourceData.dim == RESOURCE_DIMENSION_TEXTURECUBEARRAY)
+  {
+    textureDecl = "TextureCubeArray";
+    texdim = 4;
+    offsetDim = 3;
+    useOffsets = false;
+  }
+  else
+  {
+    RDCERR("Unsupported resource type %d in sample operation", resourceData.dim);
+  }
+
+  {
+    char *typeStr[NUM_RETURN_TYPES] = {
+        "",    // enum starts at ==1
+        "unorm float",
+        "snorm float",
+        "int",
+        "uint",
+        "float",
+        "__",    // RETURN_TYPE_MIXED
+        "double",
+        "__",    // RETURN_TYPE_CONTINUED
+        "__",    // RETURN_TYPE_UNUSED
+    };
+
+    // obviously these may be overly optimistic in some cases
+    // but since we don't know at debug time what the source texture format is
+    // we just use the fattest one necessary. There's no harm in retrieving at
+    // higher precision
+    DXGI_FORMAT fmts[NUM_RETURN_TYPES] = {
+        DXGI_FORMAT_UNKNOWN,    // enum starts at ==1
+        DXGI_FORMAT_R32G32B32A32_FLOAT, DXGI_FORMAT_R32G32B32A32_FLOAT,
+        DXGI_FORMAT_R32G32B32A32_SINT, DXGI_FORMAT_R32G32B32A32_UINT, DXGI_FORMAT_R32G32B32A32_FLOAT,
+        DXGI_FORMAT_UNKNOWN,    // RETURN_TYPE_MIXED
+
+        // should maybe be double, but there is no double texture format anyway!
+        // spec is unclear but I presume reads are done at most at float
+        // precision anyway since that's the source, and converted to doubles.
+        DXGI_FORMAT_R32G32B32A32_FLOAT,
+
+        DXGI_FORMAT_UNKNOWN,    // RETURN_TYPE_CONTINUED
+        DXGI_FORMAT_UNKNOWN,    // RETURN_TYPE_UNUSED
+    };
+
+    char buf[64] = {0};
+    StringFormat::snprintf(buf, 63, "%s4", typeStr[resourceData.retType]);
+
+    if(retFmt == DXGI_FORMAT_UNKNOWN)
+    {
+      funcRet = buf;
+      retFmt = fmts[resourceData.dim];
```

Comment:
Found a typo here while testing - this should be indexed by `resourceData.retType` not `resourceData.dim`.

Reply:
Oops, thanks for catching. Fixed.